src/main/java/uk/gov/companieshouse/api/accounts/validation/CurrentPeriodValidator.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.companieshouse.api.accounts.validation;
+
+
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.accounts.model.validation.Errors;
+import uk.gov.companieshouse.api.accounts.model.rest.CurrentPeriod;
+import uk.gov.companieshouse.api.accounts.model.rest.FixedAssets;
+
+import javax.validation.Valid;
+
+/**
+ * Validates Current Period
+ */
+@Component
+public class CurrentPeriodValidator extends BaseValidator {
+
+    String CURRENT_PERIOD_PATH = "$.current_period";
+    String BALANCE_SHEET_PATH = CURRENT_PERIOD_PATH + ".balance_sheet";
+    String FIXED_ASSETS_PATH = BALANCE_SHEET_PATH + ".fixed_assets";
+    String TOTAL_PATH = FIXED_ASSETS_PATH + ".total";
+
+    public Errors validateCurrentPeriod(@Valid CurrentPeriod currentPeriod, Errors errors) {
+
+        if (currentPeriod.getBalanceSheet() != null) {
```

Comment:
Null check validation occurs via the `@NotNull(message="MANDATORY_ELEMENT_MISSING")` annotation on the CurrentPeriod class no need to repeat yourself here.